de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/grid/GridViewPresenterImpl.java

```diff
@@ -483,8 +499,55 @@ public void onSuccess(ShareResourceLinkDialog result) {
                 }
             });
         } else {
-            diskResourceService.createDataLinks(Arrays.asList(toBeShared.getPath()),
-                                                new CreateDataLinksCallback());
+            diskResourceService.listDataLinks(diskResourceUtil.asStringPathList(Arrays.asList(toBeShared)),
+                                              new DataCallback<String>() {
+
+                                                  @Override
+                                                  public void onFailure(Integer statusCode,
+                                                                        Throwable exception) {
+                                                      ErrorHandler.post(appearance.createDataLinksError(),
+                                                                        exception);
+                                                  }
+
+                                                  @Override
+                                                  public void onSuccess(String result) {
+                                                      List<DataLink> dlList = processResult(result);
+                                                      if (dlList == null || dlList.isEmpty()) {
+                                                          diskResourceService.createDataLinks(Arrays.asList(
+                                                                  toBeShared.getPath()),
+                                                                                              new CreateDataLinksCallback());
+                                                      } else {
+                                                          showPublicLink(dlList);
+                                                      }
+
+                                                  }
+
+                                                  private List<DataLink> processResult(String result) {
```

Comment:
I actually wanted to re-use the entire `ListDataLinksCallback` but it had `Tree` component in it.  Let me see if I refactor some parts of callback into a converter.